notes.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"accounts-service/models"
+	accountsv1 "accounts-service/protorepo/noted/accounts/v1"
+	"accounts-service/validators"
+	"context"
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (srv *groupsAPI) AddGroupNote(ctx context.Context, in *accountsv1.AddGroupNoteRequest) (*accountsv1.AddGroupNoteResponse, error) {
+	err := validators.ValidateAddGroupNote(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to validate add member request")
+	}
+
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = srv.memberRepo.Get(ctx, &models.MemberFilter{AccountID: &accountId, GroupID: &in.GroupId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to get member from group_id")
+	}
+
```

Comment:
Maybe we should check that the note exists within the notes-service? And that the user is the author of said note.

Or **Option 2** which I like better: We have a `POST /notes` endpoint on the api gateway which takes care of calling `CreateNote` in the notes-service and then calls `AddGroupNote` with the said note. And therefore `AddGroupNote` would become an internal endpoint which is only called by the API gateway. LMKWYT.

Reply:
Yes, i had the second option in mind